src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs

```diff
@@ -313,7 +313,6 @@ public ReferencedAssemblyIdentity(AssemblyIdentity identity, MetadataReference r
                     {
                         case MetadataImageKind.Assembly:
                             var assemblyMetadata = (AssemblyMetadata)metadata;
-                            WeakList<IAssemblySymbol> cachedSymbols = assemblyMetadata.CachedSymbols;
```

Comment:
>WeakList<IAssemblySymbol> cachedSymbols = assemblyMetadata.CachedSymbols [](start = 28, length = 72)

Was this code accessing the field without taking the SymbolCacheAndReferenceManagerStateGuard lock?